### PR TITLE
Ide options

### PR DESCRIPTION
**Pull request by tmat**
Moves EditorConfigOptionsEnumerator to Features so that it can enumerate editorconfig options defined in Features layer (options used by analyzers implemented only in Features).

Converts options passed to analyzers/code fixes via `IdeAnalyzerOptions` to editorconfig options.

Unsupported:
- `dotnet_unsupported_report_invalid_placeholders_in_string_dot_format_calls`
- `dotnet_unsupported_detect_and_offer_editor_features_for_probable_json_strings`
- `dotnet_unsupported_report_invalid_json_patterns`
- `dotnet_unsupported_report_invalid_regex_patterns`

Supported:
- `dotnet_prefer_system_hash_code`

Follow up on https://github.com/dotnet/roslyn/pull/74038.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-ide @sharwell @CyrusNajmabadi PTAL, Please, skip the first commit -- it's the same as https://github.com/dotnet/roslyn/pull/74038.

File: src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexOptionsStorage.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.Features.EmbeddedLanguages.RegularExpressions.LanguageServices;
+
+internal static class RegexOptionsStorage
+{
+    public static PerLanguageOption2<bool> ReportInvalidRegexPatterns = new(
+        "dotnet_unsupported_report_invalid_regex_patterns",
+        defaultValue: true,
+        isEditorConfigOption: true);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexOptionsStorage.cs:15]: and yjust to make sure.  a user can still turn this  on/off in the VS ui?

File: src/VisualStudio/CSharp/Test/EditorConfigSettings/DataProvider/DataProviderTests.cs [hunk 2]
```diff
@@ -106,6 +106,7 @@ public void TestGettingAnalyzerSettingsProviderWorkspaceServiceAsync()
         }
 
         [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/62937")]
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Test/EditorConfigSettings/DataProvider/DataProviderTests.cs:109]:
```suggestion
        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/62937")]
```

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexOptionsStorage.cs (hunk 1 above)
Comment (tmat) [reply src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexOptionsStorage.cs:15]: Yes. Via Tools > Options.

Comment (tmat) [reply src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexOptionsStorage.cs:15]: (but not via editorconfig UI)

**Pull request merged**